### PR TITLE
Feature/pro 1434 subclustering

### PR DESCRIPTION
**Pull request by boba-and-beer**

### CODE REVIEW

File: relevanceai/clusterer/clusterer.py [hunk 1]
```diff
@@ -1086,11 +1103,201 @@ def fit_predict(self, X):
...
+                + f"https://cloud.relevance.ai/dataset/{self.dataset_id}/deploy/recent/cluster"
+            )
+
+    def subfit_predict_update(
+        self,
+        dataset,
+        vector_fields: list = None,
+        filters: list = None,
+        verbose: bool = False,
+    ):
```
Comment (ofrighil) [inline relevanceai/clusterer/clusterer.py:1118]: This method is missing a docstring

File: relevanceai/clusterer/clusterer.py [hunk 2]
```diff
@@ -1086,11 +1103,201 @@ def fit_predict(self, X):
...
+            The dataset to call fit predict update on
+        vector_fields: list
+            The list of vector fields
+        filters: list
+            The list of filters
+
+        Example
+        ----------
+
+        .. code-block::
```
Comment (ofrighil) [inline relevanceai/clusterer/clusterer.py:1190]: Is this code block suppose to be filled?

File: relevanceai/clusterer/clusterer.py [hunk 3]
```diff
@@ -1086,11 +1103,201 @@ def fit_predict(self, X):
...
+                + f"https://cloud.relevance.ai/dataset/{self.dataset_id}/deploy/recent/cluster"
+            )
+
+    def subfit_predict_documents(
+        self,
+        vector_fields,
+        filters: list,
+        cluster_ids: list = None,
+        verbose: bool = True,
+    ):
```
Comment (ofrighil) [inline relevanceai/clusterer/clusterer.py:1235]: Missing docstring. Additionally, the function signature should be changed to contain:
```
...
filters: Optional[list] = None,
cluster_ids: Optional[list] = None,
...
```

File: relevanceai/clusterer/clusterer.py [hunk 4]
```diff
@@ -1086,11 +1103,201 @@ def fit_predict(self, X):
...
+
+        print("Getting unique cluster IDs...")
+        if not cluster_ids:
+            unique_clusters = self.unique_cluster_ids(alias=self.parent_alias)
+        else:
+            unique_clusters = cluster_ids
+
+        for i, unique_cluster in enumerate(tqdm(unique_clusters)):
+            cluster_filters = filters.copy()
+            cluster_filters += [
```
Comment (ofrighil) [inline relevanceai/clusterer/clusterer.py:1249]: Nit: I think append is more idiomatic. Also, is there a need to make a copy of `filters`?

File: relevanceai/config.ini [hunk 5]
```diff
@@ -28,3 +28,6 @@ show_dashboard_link = True
 [mixpanel]
 is_tracking_enabled = True
 write_key = TWlZWXF6dFVMOGxyRlJMZjJ0YmhmNGZmSlFuZEp3R0M=
```
Comment (ofrighil) [inline relevanceai/config.ini:30]: Should this be public?

**Review by ofrighil — APPROVED** [review]

File: relevanceai/clusterer/clusterer.py (hunk 4 above)
Comment (boba-and-beer) [reply relevanceai/clusterer/clusterer.py:1249]: Yep. Here is a code snippet to try out. 

```
a = ['a', 'b']
b = a
b += ['c']
a
```

Comment (ofrighil) [reply relevanceai/clusterer/clusterer.py:1249]: Ok, I see your point. I didn't think of the case in which a user would reuse a filter used in the argument.

**Pull request merged**